File: FRONT/frontend/proyecto-Gestion-Ambiental/src/app/models/Reporte.ts
//Autor:Eduardo Chavez
export interface Reporte {
    id_reporte?: number;
    descripcion: string;
    fecha?: string | Date; 
    observaciones?: string;
    aprobado?: boolean;
        
    id_actividad?: number;
    id_user?: number;       
    id_iniciativa?: number; 

    id_actividadNavigation?: any;
    id_iniciativaNavigation?: any;
    id_userNavigation?: any;
}

File: BACK/backend/Proyecto-Gestion-Ambiental/databasefirst/Models/Reporte.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace databasefirst.Models;

public partial class Reporte
{
    public int id_reporte { get; set; }
    public string? descripcion { get; set; }
    public DateTime? fecha { get; set; }
    public string? observaciones { get; set; }
    public bool? aprobado { get; set; }

    public int? id_actividad { get; set; }
    public int id_user { get; set; }     
    public int? id_iniciativa { get; set; } 


    [ForeignKey("id_actividad")]
    public virtual Actividad? id_actividadNavigation { get; set; }

    [ForeignKey("id_user")]
    public virtual Usuario? id_userNavigation { get; set; }

    [ForeignKey("id_iniciativa")]
    public virtual Iniciativa? id_iniciativaNavigation { get; set; }
}